### PR TITLE
DO NOT MERGE, DISCUSSION ONLY : [NS-523] Boto Auth switch over recipe

### PR DESCRIPTION
**Pull request by sandeepmjay**
Boto authentication using app_role and STS. AMI config changes for IAM instance profiles, Buildspec role, smart_open version bump.

The code uses the Boto session to create STS and updated smart_open. 
The issue from NS-532 persists on both boac-dev and the boa-test(environment I m running). This is independent of the auth scheme used by Boto.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ nltk==3.4.0
 psycopg2==2.7.7
 requests==2.21.0
 simplejson==3.16.0
-smart-open==1.8.0
+smart-open==1.8.3
```
Comment (johncrossman) [inline requirements.txt]: @sandeepmjay  `boto3` upgrade not needed?

Comment (sandeepmjay) [reply requirements.txt]: nope.

File: .ebextensions/00_ami.config [hunk 2]
```diff
@@ -47,7 +47,7 @@ option_settings:
   aws:elbv2:listener:443:
     ListenerEnabled: 'true'
     Protocol: HTTPS
-    SSLCertificateArns: arn:aws:acm:us-west-2:697877139013:certificate/f81da03d-f9c9-40f2-ad99-521e051203f7
+    SSLCertificateArns: arn:aws:acm:us-west-2:697877139013:certificate/a50eaef4-d167-40eb-a834-7e071d88650f
```
Comment (sandeepmjay) [inline .ebextensions/00_ami.config]: Using cloud-lrs cert to test things independent of boac deploys.

File: boac/externals/s3.py [hunk 3]
```diff
@@ -54,13 +64,30 @@ def put_file_to_s3(bucket, key, path_to_file):
...
+    sts_client = boto3.client('sts')
+    role_arn = app.config['AWS_APP_ROLE_ARN']
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName='AssumeAppRoleSession',
+    )
+
+    credentials = assumed_role_object['Credentials']
+    app.logger.info(f'credentials: {credentials}')
+    return credentials
```
Comment (sandeepmjay) [inline boac/externals/s3.py]: TODO: Add a single block session after all test pass. Don't mind the redundant boto session for now.

**Pull request closed without merging**